File: src/main/java/com/jocoos/mybeautip/domain/notice/persistence/domain/NoticeFile.java
package com.jocoos.mybeautip.domain.notice.persistence.domain;

import com.jocoos.mybeautip.domain.file.code.FileType;
import lombok.*;

import javax.persistence.*;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class NoticeFile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private FileType type;

    @Column(nullable = false)
    private String file;

    @Setter
    @ManyToOne
    private Notice notice;

    public void editFile(String file) {
        this.file = file;
    }
}

File: src/main/kotlin/com/mybeautip/domain/notice/persistence/domain/NoticeFile.kt
package com.mybeautip.domain.notice.persistence.domain

import com.jocoos.mybeautip.global.config.jpa.PrimaryKeyEntity
import com.mybeautip.global.code.FileType
import javax.persistence.Column
import javax.persistence.FetchType
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne

//@Entity
class NoticeFile (
    @Column(nullable = false) var file: String,
    @Column(nullable = false) var type: FileType = FileType.IMAGE
) : PrimaryKeyEntity() {

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "notice_id", nullable = false)
    var notice: Notice? = null

}